Navigator/boards: introduce RAM_CONSTRAINED_SYSTEM for reduced mission length

sfuhrer opened the pull request:
### Solved Problem
The current limit of mission items is quite low. For large survey missions it's possible to exceed the limit of 500. This limit was introduce when the ability to store [missions on RAM was added](https://github.com/PX4/PX4-Autopilot/pull/18827), as RAM is much more limited than the SDcard in terms of available storage space, and a full RAM leads to a hard-fault. 

### Solution
Add `#define RAM_CONSTRAINED_SYSTEM` to specify if a board has a constrained RAM space, in which case the limit is left at 500. If that define is not set then open up the limit to 1000 (arbitrary number). 

Side question: is MEMORY_CONSTRAINED_SYSTEM even set anywhere?

### Changelog Entry
For release notes:
```
Improvement: Navigator/boards: introduce `#define RAM_CONSTRAINED_SYSTEM` for reduced mission length
```

### Alternatives
@MaEtUgR mentioned that we could consider reducing the amount of data such a mission write needs, at least if stored on RAM. Binding the max mission size to `SYS_DM_BACKEND` is likely not feasible. 

This is just an initial draft PR to kick off some discussion, I didn't go through all the boards and labeled them correctly with `RAM_CONSTRAINED_SYSTEM`.

PetervdPerk-NXP commented:
How about making it a Kconfig symbol with a help text explaining what it does?

dagar reviewed (changes requested):
We don't need a generic "RAM_CONSTRAINED_SYSTEM", we can simply do a real kconfig level `NUM_MISSIONS_SUPPORTED` for navigator/dataman.

sfuhrer commented:
@dagar @PetervdPerk-NXP thanks for the inputs, I made a kconfig option now. 
For now I left it by default to be constrained to 500, and raise it for boards with more RAM (>1kb) to 1000 (I likely missed some boards). We could also flip it around and by default have it at 1000, and lower it for boards with RAM <1kb. Thoughts?

MaEtUgR commented on src/modules/navigator/Kconfig:
```diff
@@ -27,3 +27,12 @@ menuconfig NAVIGATOR_ADSB
+	default 500
+	depends on MODULES_NAVIGATOR
+	---help---
+		This number is board specific, as the RAM size varies and with it how many mission
+		items can be saved on it. At the moment we do not distinguish between saving on
+		RAM and SD card, the same limit applies to both.
```
```suggestion
		This limit always applies if mission items are stored on the SD card or in RAM. It should be set
		adequately per boards such that if the items are stored in RAM they still fit the available memory.
		The runtime parameter to configure the storage option is `SYS_DM_BACKEND`.
```

MaEtUgR reviewed (approved):
Removing the strange hardcoded profiles makes sense anyways. Also having it configurable per board is probably a necessary evil to accommodate for low RAM chips but still not be limited for big missions.

I'd suggest to clearly distinguish between where the items are stored but that's a more involved change.

I have some minor suggestions for the help text. Feel free to mix what you consider understandable.

PetervdPerk-NXP reviewed (approved):
LGTM in the future we could simplify this if more cases occurs by using the kconfig `default if` syntax

Example from zephyr kconfig
```
 config SHELL_STACK_SIZE 
 	int "Shell thread stack size" 
 	default 3168 if OPENTHREAD_SHELL 
 	default 3072 if 64BIT 
 	default 2048 if MULTITHREADING 
 	default 0 if !MULTITHREADING 
```

sfuhrer commented:
I've rebased and cleaned up the commits otherwise, so this would be good to go unless we want to include https://github.com/PX4/PX4-Autopilot/pull/24252#pullrequestreview-2699009430.

PetervdPerk-NXP commented:
> > LGTM in the future we could simplify this if more cases occurs by using the kconfig `default if` syntax
> > Example from zephyr kconfig
> > ```
> >  config SHELL_STACK_SIZE 
> >  	int "Shell thread stack size" 
> >  	default 3168 if OPENTHREAD_SHELL 
> >  	default 3072 if 64BIT 
> >  	default 2048 if MULTITHREADING 
> >  	default 0 if !MULTITHREADING 
> > ```
> 
> @PetervdPerk-NXP that sounds interesting, which if() could we use to find boards with a certain RAM size?

I was thinking about defining certain device classes.

sfuhrer commented:
@PetervdPerk-NXP was wondering if you could help me with the [one CI failure](https://github.com/PX4/PX4-Autopilot/actions/runs/13951772111/job/39052979953?pr=24252) we now get here..
`nxp_mr-canhubk3_default` fails as the only board due to `CONFIG_NUM_MISSION_ITMES_SUPPORTED' was not declared in this scope`.

PetervdPerk-NXP commented:
Looks to be a dependency issue your kconfig symbol `NUM_MISSION_ITMES_SUPPORTED` depends on navigator, however the `nxp_mr-canhubk3_default` doesn't have navigator enabled. 

I just enabled navigator on the k3 target by default, I think CI should succeed now.
